snaxc/accelerators/dispatching.py

```diff
@@ -13,6 +13,40 @@ class SupportedKernel:
     kernel_type: type[KernelOp]
     operand_types: Iterable[Attribute]
 
+    def is_same_kernel(self, kernel_op: Operation | None) -> bool:
+        """
+        Check if the kernel operation matches the supported kernel type and operand types.
+        """
+        if kernel_op is None:
+            return False
+        if not isinstance(kernel_op, KernelOp):
+            return False
+        return self._check_kernel_type(kernel_op) and self._check_attribute_types(kernel_op)
+
+    def _check_kernel_type(self, kernel_op: KernelOp) -> bool:
+        """
+        Check if the kernel operation is of the supported type.
+        """
+        return isinstance(kernel_op, self.kernel_type)
+
+    def _check_attribute_types(self, kernel_op: KernelOp) -> bool:
+        """
+        Check if the attributes of the kernel operation match the expected types.
+        """
+        expected_types = list(self.operand_types)
+        i = 0
+        for operand_type in kernel_op.operand_types:
+            if operand_type != expected_types[i]:
+                return False
+            i += 1
+        for result_type in kernel_op.result_types:
+            if result_type != expected_types[i]:
+                return False
+            i += 1
+        if i != len(expected_types):
+            return False
+        return True
```

Comment:
I like this functionality, but it can also be written in 3 lines I think?
```suggestion
    def is_same_kernel(self, kernel_op: Operation | None) -> bool:
        """
        Check if the kernel operation matches the supported kernel type and operand types.
        """
        if not isinstance(kernel_op, self.kernel_type):
            return False
        return list(self.operand_types) == [*kernel_op.operand_type, *kernel_op.result_types]
```